File: backend/models/Booking.js
import { DataTypes } from 'sequelize';
import { sequelize } from '../config/database.js';

const Booking = sequelize.define('Booking', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  userId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  serviceId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'services',
      key: 'id'
    }
  },
  providerId: {
    type: DataTypes.INTEGER,
    allowNull: true,
    defaultValue: null,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  bookingDate: {
    type: DataTypes.DATE,
    allowNull: false
  },
  bookingTime: {
    type: DataTypes.STRING,
    allowNull: false
  },
  address: {
    type: DataTypes.STRING,
    allowNull: false
  },
  totalAmount: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM('pending', 'confirmed', 'in-progress', 'completed', 'cancelled'),
    defaultValue: 'pending'
  },
  paymentStatus: {
    type: DataTypes.ENUM('pending', 'paid', 'refunded'),
    defaultValue: 'pending'
  }
}, {
  tableName: 'bookings',
  timestamps: true,
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
});

export default Booking;

File: backend_fastapi/models/booking.py
from typing import Optional
from datetime import datetime
from sqlmodel import SQLModel, Field, Relationship
from enum import Enum

class BookingStatus(str, Enum):
    PENDING = "pending"
    CONFIRMED = "confirmed"
    IN_PROGRESS = "in-progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"

class BookingBase(SQLModel):
    userId: int = Field(foreign_key="user.id")
    serviceId: int = Field(foreign_key="service.id")
    providerId: Optional[int] = Field(default=None, foreign_key="user.id")
    bookingDate: datetime
    bookingTime: str
    address: str
    totalAmount: float
    status: BookingStatus = Field(default=BookingStatus.PENDING)
    paymentStatus: str = Field(default="pending")

class Booking(BookingBase, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    
    user: "User" = Relationship(back_populates="bookings_as_customer", sa_relationship_kwargs={"primaryjoin": "foreign(Booking.userId) == User.id"})
    provider: Optional["User"] = Relationship(back_populates="bookings_as_provider", sa_relationship_kwargs={"primaryjoin": "foreign(Booking.providerId) == User.id"})
    service: "Service" = Relationship(back_populates="bookings")

from .user import User
from .service import Service
